code/modules/surgery/organs/lungs.dm

```diff
@@ -433,6 +435,11 @@
 		else
 			SEND_SIGNAL(owner, COMSIG_CLEAR_MOOD_EVENT, "smell")
 
+		if (n2o_euphoria || healium_euphoria)
+			SEND_SIGNAL(owner, COMSIG_ADD_MOOD_EVENT, "chemical_euphoria", /datum/mood_event/chemical_euphoria)
+		else 
+			SEND_SIGNAL(owner, COMSIG_CLEAR_MOOD_EVENT, "chemical_euphoria")
+
```

Comment:
Since these default to false, doesn't that mean that if the prob()s don't run, we'll clear the mood event? true and false might not be enough to represent the information here. Maybe defines? tho that sounds like it might get messy fast